editor/editor_inspector.cpp

```diff
@@ -382,12 +382,24 @@ void EditorProperty::update_property() {
 	GDVIRTUAL_CALL(_update_property);
 }
 
+void EditorProperty::_set_read_only(bool p_read_only) {
+	if (get_script_instance() && get_script_instance()->has_method("_set_read_only")) {
+		get_script_instance()->call("_set_read_only", p_read_only);
+	}
+}
+
+void EditorProperty::set_global_read_only(bool p_read_only) {
```

Comment:
No, this does not make a lot of sense, it's not up the EditorProperty class to handle something that's part of the EditorInspector.

It's up to the EditorInspector itself to handle this, setting all children EditorProperty to read_only when the inspector itself is set to read_only, or only the correct ones when it's not.

Reply:
If there is only one flag, it is not possible to decide whether it is readonly by the inspector or property itself. For example, in the use case #30623, some properties will be readonly individually, but when the inspector becomes `readonly==true` due to other factors, and then the inspector itself becomes `readonly==false`, the readonly flag of all properties is set to `false`, so the individual property's readonly becomes meaningless.

Reply:
> but when the inspector becomes readonly==true due to other factors, and then the inspector itself becomes readonly==false, the readonly flag of all properties is set to false, so the individual property's readonly becomes meaningless.

No, that's not how it should work. When changing the inspector's read_only status, you can simply re-create the whole tree to solve the problem by calling `update_tree()`. It's not the most efficient way but performance here is not a problem (changing the inspector read_only status is not something done very often). `update_tree()` should then set the property to read_only when either the property is set to read only or if the inspector is set to read_only.